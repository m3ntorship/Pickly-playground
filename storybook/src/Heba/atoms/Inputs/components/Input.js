import React from "react";

const Input = (props) => {
  const {
    leftIcon,
    rightIcon,
    prefix,
    prefixDrop,
    variant,
    disabled,
    click,
    setHover,
  } = props;
  return (
    <input
      className={`w-72 h-3xl   ${
        leftIcon
          ? !rightIcon
            ? "pl-9"
            : "pl-m"
          : prefix
          ? "pl-3xl"
          : prefixDrop
          ? "pl-4xl"
          : "pl-m"
      } text-sm text-grey-shd1 font-normal border rounded-md hover:border-grey-shd2 focus:text-dark-grey ${
        variant === "error"
          ? "border-error focus:border-error"
          : variant === "success"
          ? "border-success focus:border-success"
          : "border-grey-shd5 focus:border-dark"
      } focus:outline-none ${
        disabled ? "pointer-events-none opacity-50 border border-grey-shd5" : ""
      }`}
      type="text"
      placeholder="What do you ask about?"
      onClick={click}
      onMouseEnter={() => (leftIcon ? setHover(true) : null)}
      onMouseLeave={() => (leftIcon ? setHover(false) : null)}
      style={{ width: "580px" }}
    />
  );
};

export default Input;
