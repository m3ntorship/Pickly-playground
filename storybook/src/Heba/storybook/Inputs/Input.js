import React from "react";
export const Input_Optins = {
  Case: {
    Normal: "normal",
    Success: "success",
    Failure: "failure",
  },
};
export const Input = ({ casetype }) => {
  {
    if (casetype === Input_Optins.Case.Normal) {
      return (
        <div className="w-80 h-10 flex-col justify-start">
          <div className="flex items-center pl-1 align-middle rounded-md border border-dark">
            <input
              type="text"
              id="text2"
              placeholder="Enter text"
              className="w-80 h-10 text-sm outline-none"
            />
            <svg
              className="ml-0 mr-2"
              width="14"
              height="14"
              viewBox="0 0 14 14"
              fill="none"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M7.00016 0.333313C3.32416 0.333313 0.333496 3.32398 0.333496 6.99998C0.333496 10.676 3.32416 13.6666 7.00016 13.6666C10.6762 13.6666 13.6668 10.676 13.6668 6.99998C13.6668 3.32398 10.6762 0.333313 7.00016 0.333313ZM9.80483 8.86198L8.86216 9.80465L7.00016 7.94265L5.13816 9.80465L4.1955 8.86198L6.0575 6.99998L4.1955 5.13798L5.13816 4.19531L7.00016 6.05731L8.86216 4.19531L9.80483 5.13798L7.94283 6.99998L9.80483 8.86198Z"
                fill="#8D9A9E"
              ></path>
            </svg>
          </div>
        </div>
      );
    }
    if (casetype === Input_Optins.Case.Success) {
      return (
        <div className="w-80 h-10 flex-col justify-start">
          <div className="flex items-center pl-1 align-middle rounded-md border border-success">
            <input
              type="text"
              id="text2"
              placeholder="Enter text"
              className="w-80 h-10 text-sm outline-none"
            />
            <svg
              className="ml-0 mr-3"
              width="16"
              height="16"
              viewBox="0 0 16 16"
              fill="none"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M6.66681 10.3906L4.47148 8.19527L3.52881 9.13794L6.66681 12.2759L13.1381 5.8046L12.1955 4.86194L6.66681 10.3906Z"
                fill="#07B255"
              ></path>
            </svg>
          </div>
        </div>
      );
    }
    if (casetype === Input_Optins.Case.Failure) {
      return (
        <div className="w-80 h-10 flex-col justify-start">
          <div className="flex items-center align-middle rounded-md pl-1 border border-error">
            <input
              type="text"
              id="text2"
              placeholder="Enter text"
              className="w-80 h-10 text-sm outline-none"
            />

            <svg
              className="ml-0 mr-3"
              width="2"
              height="8"
              viewBox="0 0 2 8"
              fill="none"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0.333496 0.666626H1.66683V5.33329H0.333496V0.666626ZM0.333496 5.99996H1.66683V7.33329H0.333496V5.99996Z"
                fill="#FE544A"
              ></path>
            </svg>
          </div>
        </div>
      );
    }
  }
};
