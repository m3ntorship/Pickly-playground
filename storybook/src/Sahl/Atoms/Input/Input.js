import React, { useState } from "react";
import PropTypes from "prop-types";
import SelectDropDown from "../SelectDropDown/SelectDropDown";

// fix: disabled doesn't work now
// fix: onClick on close svg doesn't work

export default function Input(props) {
  const {
    label = "",
    error = false,
    success = false,
    disabled = false,
    type = "text",
    leftIcon = "",
    name,
    placeholder = "",
    dropDownPrefix = false,
    dropDownOptions = [],
  } = props;

  let borderColor = "border-grey-shd5";
  let statusIcon = null;

  // State
  // input value in case of text/textarea
  const [value, setValue] = useState("");
  // input value in case of file
  const [file, setFile] = useState(null);
  // Handlers
  const onInputChange = (e) => {
    setValue(e.target.value);
  };

  if (error) {
    borderColor = "border-error";
    statusIcon = (
      <svg
        className="w-4 h-4 text-error"
        fill="none"
        stroke="currentColor"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          d="M10 14l2-2m0 0l2-2m-2 2l-2-2m2 2l2 2m7-2a9 9 0 11-18 0 9 9 0 0118 0z"
        ></path>
      </svg>
    );
  }

  if (success) {
    borderColor = "border-success";
    statusIcon = (
      <svg
        className="w-4 h-4 text-success"
        fill="none"
        stroke="currentColor"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          d="M9 12l2 2 4-4m6 2a9 9 0 11-18 0 9 9 0 0118 0z"
        ></path>
      </svg>
    );
  }

  // Markup
  let markup = null;
  switch (type) {
    case "text":
      markup = (
        <div className="w-full">
          <label htmlFor={name} className="block mb-2 text-2xs md:text-xs ">
            {label}
          </label>
          <div
            className={`bg-white flex items-center h-8 rounded-md px-3 py-4 border ${borderColor} hover:border-grey-shd3 group focus:border-dark-grey ${
              disabled && "opacity-40 cursor-not-allowed"
            }`}
          >
            {leftIcon === "home" ? (
              <svg
                className="w-6 h-4 border-r border-grey-shd5 pr-2 br mr-4 font-light"
                fill="none"
                stroke="currentColor"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  strokeLinecap="round"
                  strokeLinejoin="round"
                  strokeWidth="2"
                  d="M3 12l2-2m0 0l7-7 7 7M5 10v10a1 1 0 001 1h3m10-11l2 2m-2-2v10a1 1 0 01-1 1h-3m-6 0a1 1 0 001-1v-4a1 1 0 011-1h2a1 1 0 011 1v4a1 1 0 001 1m-6 0h6"
                ></path>
              </svg>
            ) : leftIcon === "caption" ? (
              <svg
                className="w-6 h-4 border-r border-grey-shd5 pr-2 br mr-4 font-light"
                viewBox="0 0 24 24"
                fill="none"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 4C0 1.79086 1.79086 0 4 0H20C22.2091 0 24 1.79086 24 4V20C24 22.2091 22.2091 24 20 24H4C1.79086 24 0 22.2091 0 20V4Z"
                  fill="#F2F9FF"
                />
                <path
                  d="M14.085 14.4023H9.91504L8.97852 17H7.625L11.4258 7.04688H12.5742L16.3818 17H15.0352L14.085 14.4023ZM10.3115 13.3223H13.6953L12 8.66699L10.3115 13.3223Z"
                  fill="#8D9A9E"
                />
              </svg>
            ) : null}
            {dropDownPrefix && (
              <SelectDropDown
                options={dropDownOptions}
                className="border-r border-grey-shd5 br mr-4"
              />
            )}
            <input
              id={name}
              disabled={disabled}
              type={type}
              name={name}
              placeholder={placeholder}
              value={value}
              onChange={onInputChange}
              className={`outline-none block border-0 w-full mr-auto text-2xs md:text-xs ${
                disabled && "cursor-not-allowed"
              }`}
            />
            {/* Typing Icon... */}
            <svg
              className={`w-4 h-4 ${
                value.length > 0 ? "block" : "hidden"
              } mr-4 cursor-pointer`}
              viewBox="0 0 24 24"
              fill="none"
              onClick={() => {
                setValue("");
              }}
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 2C6.486 2 2 6.486 2 12C2 17.514 6.486 22 12 22C17.514 22 22 17.514 22 12C22 6.486 17.514 2 12 2ZM16.207 14.793L14.793 16.207L12 13.414L9.207 16.207L7.793 14.793L10.586 12L7.793 9.207L9.207 7.793L12 10.586L14.793 7.793L16.207 9.207L13.414 12L16.207 14.793Z"
                fill="#5B6366"
              />
            </svg>
            {statusIcon}
            {}
          </div>
        </div>
      );
      break;

    case "file":
      markup = (
        <button className="relative border-2 border-dashed border-accent bg-grey-shd7 rounded-md text-accent focus:outline-none flex justify-center items-center mx-2 sm:mx-0 w-full h-14">
          <input
            disabled={disabled}
            type={type}
            name={name}
            value={file}
            onChange={(e) => {
              setFile(e.target.value);
            }}
            className="cursor-pointer opacity-0 absolute w-full h-full border-2 border-success"
          />
          <svg
            className="mr-1 w-4 h-4 fill-current"
            viewBox="0 0 24 24"
            fill="currentColor"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M4 5H17V12H19V5C19 3.897 18.103 3 17 3H4C2.897 3 2 3.897 2 5V17C2 18.103 2.897 19 4 19H12V17H4V5Z"
              fill="currentColor"
            />
            <path d="M8 11L5 15H16L12 9L9 13L8 11Z" fill="currentColor" />
            <path
              d="M19 14H17V17H14V19H17V22H19V19H22V17H19V14Z"
              fill="currentColor"
            />
          </svg>
          <span className="text-link">Upload one or multiple images</span>
        </button>
      );
      break;

    default:
      break;
  }

  return <div className="bg-success h-28">{markup}</div>;
}

Input.propTypes = {
  label: PropTypes.string,
  type: PropTypes.oneOf(["textarea", "file", "text"]),
  success: PropTypes.bool,
  error: PropTypes.bool,
  disabled: PropTypes.bool,
  leftIcon: PropTypes.string,
  dropDownPrefix: PropTypes.bool,
  dropDownOptions: PropTypes.array,
};
