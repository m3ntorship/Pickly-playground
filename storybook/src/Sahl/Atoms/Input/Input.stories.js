import React from "react";

import Input from "./Input";

const defaultExport = {
  title: "Sahl/Atoms/Input",
  component: Input,
};

export default defaultExport;

const Template = (args) => <Input {...args} />;

export const Default = Template.bind({});
Default.args = {
  label: "Default",
  type: "text",
  error: false,
  success: false,
  leftIcon: "",
  disabled: false,
  placeholder: "This is input...",
  dropDownOptions: ["+20", "+984", "+44"],
};
