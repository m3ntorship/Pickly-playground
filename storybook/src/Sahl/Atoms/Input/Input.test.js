import React from "react";
import renderer from "react-test-renderer";
import Input from "./Input";

describe("Input Component", () => {
  it("should render a default input wihtout any props ", () => {
    const tree = renderer.create(<Input />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render disabled when disabled is passed ", () => {
    const tree = renderer.create(<Input disabled />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render with close icon to right when typing only ", () => {
    const tree = renderer.create(<Input />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render with check icon to right when passing sccess", () => {
    const tree = renderer.create(<Input success />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render with x icon to right when passing error", () => {
    const tree = renderer.create(<Input error />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render with left home icon when passing leftIcon=home", () => {
    const tree = renderer.create(<Input leftIcon="home" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render with empty select dropdown at left when passing dropDownPrefix", () => {
    const tree = renderer.create(<Input dropDownPrefix />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render with select dropdown with its options when passing dropDownPrefix & dropDownOptions=[1,2,3]", () => {
    const tree = renderer
      .create(<Input dropDownPrefix dropDownOptions={[1, 2, 3, 4]} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
