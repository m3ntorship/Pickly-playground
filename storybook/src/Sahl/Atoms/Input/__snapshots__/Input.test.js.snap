// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Component should render a default input wihtout any props  1`] = `
<div
  className="w-full"
>
  <label
    className="block mb-2 text-2xs md:text-xs "
  >
    
  </label>
  <div
    className="flex items-center h-8 rounded-md px-3 py-4 border border-grey-shd5 hover:border-grey-shd3 group focus:border-dark-grey false"
  >
    <input
      className="outline-none block border-0 w-full mr-auto text-2xs md:text-xs false"
      disabled={false}
      onChange={[Function]}
      placeholder=""
      type="text"
      value=""
    />
    <svg
      className="w-4 h-4 hidden mr-4"
      fill="none"
      onClick={[Function]}
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 2C6.486 2 2 6.486 2 12C2 17.514 6.486 22 12 22C17.514 22 22 17.514 22 12C22 6.486 17.514 2 12 2ZM16.207 14.793L14.793 16.207L12 13.414L9.207 16.207L7.793 14.793L10.586 12L7.793 9.207L9.207 7.793L12 10.586L14.793 7.793L16.207 9.207L13.414 12L16.207 14.793Z"
        fill="#5B6366"
      />
    </svg>
  </div>
</div>
`;

exports[`Input Component should render disabled when disabled is passed  1`] = `
<div
  className="w-full"
>
  <label
    className="block mb-2 text-2xs md:text-xs "
  >
    
  </label>
  <div
    className="flex items-center h-8 rounded-md px-3 py-4 border border-grey-shd5 hover:border-grey-shd3 group focus:border-dark-grey opacity-40 cursor-not-allowed"
  >
    <input
      className="outline-none block border-0 w-full mr-auto text-2xs md:text-xs cursor-not-allowed"
      disabled={true}
      onChange={[Function]}
      placeholder=""
      type="text"
      value=""
    />
    <svg
      className="w-4 h-4 hidden mr-4"
      fill="none"
      onClick={[Function]}
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 2C6.486 2 2 6.486 2 12C2 17.514 6.486 22 12 22C17.514 22 22 17.514 22 12C22 6.486 17.514 2 12 2ZM16.207 14.793L14.793 16.207L12 13.414L9.207 16.207L7.793 14.793L10.586 12L7.793 9.207L9.207 7.793L12 10.586L14.793 7.793L16.207 9.207L13.414 12L16.207 14.793Z"
        fill="#5B6366"
      />
    </svg>
  </div>
</div>
`;

exports[`Input Component should render with check icon to right when passing sccess 1`] = `
<div
  className="w-full"
>
  <label
    className="block mb-2 text-2xs md:text-xs "
  >
    
  </label>
  <div
    className="flex items-center h-8 rounded-md px-3 py-4 border border-success hover:border-grey-shd3 group focus:border-dark-grey false"
  >
    <input
      className="outline-none block border-0 w-full mr-auto text-2xs md:text-xs false"
      disabled={false}
      onChange={[Function]}
      placeholder=""
      type="text"
      value=""
    />
    <svg
      className="w-4 h-4 hidden mr-4"
      fill="none"
      onClick={[Function]}
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 2C6.486 2 2 6.486 2 12C2 17.514 6.486 22 12 22C17.514 22 22 17.514 22 12C22 6.486 17.514 2 12 2ZM16.207 14.793L14.793 16.207L12 13.414L9.207 16.207L7.793 14.793L10.586 12L7.793 9.207L9.207 7.793L12 10.586L14.793 7.793L16.207 9.207L13.414 12L16.207 14.793Z"
        fill="#5B6366"
      />
    </svg>
    <svg
      className="w-4 h-4 text-success"
      fill="none"
      stroke="currentColor"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M9 12l2 2 4-4m6 2a9 9 0 11-18 0 9 9 0 0118 0z"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
      />
    </svg>
  </div>
</div>
`;

exports[`Input Component should render with close icon to right when typing only  1`] = `
<div
  className="w-full"
>
  <label
    className="block mb-2 text-2xs md:text-xs "
  >
    
  </label>
  <div
    className="flex items-center h-8 rounded-md px-3 py-4 border border-grey-shd5 hover:border-grey-shd3 group focus:border-dark-grey false"
  >
    <input
      className="outline-none block border-0 w-full mr-auto text-2xs md:text-xs false"
      disabled={false}
      onChange={[Function]}
      placeholder=""
      type="text"
      value=""
    />
    <svg
      className="w-4 h-4 hidden mr-4"
      fill="none"
      onClick={[Function]}
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 2C6.486 2 2 6.486 2 12C2 17.514 6.486 22 12 22C17.514 22 22 17.514 22 12C22 6.486 17.514 2 12 2ZM16.207 14.793L14.793 16.207L12 13.414L9.207 16.207L7.793 14.793L10.586 12L7.793 9.207L9.207 7.793L12 10.586L14.793 7.793L16.207 9.207L13.414 12L16.207 14.793Z"
        fill="#5B6366"
      />
    </svg>
  </div>
</div>
`;

exports[`Input Component should render with empty select dropdown at left when passing dropDownPrefix 1`] = `
<div
  className="w-full"
>
  <label
    className="block mb-2 text-2xs md:text-xs "
  >
    
  </label>
  <div
    className="flex items-center h-8 rounded-md px-3 py-4 border border-grey-shd5 hover:border-grey-shd3 group focus:border-dark-grey false"
  >
    <div
      className="border-r border-grey-shd5 pr-2 br mr-4 flex items-center"
    >
      <label
        className="text-body-small text-dark-grey capitalize hidden"
      >
        :
      </label>
      <div
        className="flex items-center"
      >
        <select
          className="relative z-10 appearance-none border-none bg-transparent pr-4 m-0 w-full font-sans text-body-small capitalize cursor-pointer outline-none"
        />
        <svg
          className="w-4 h-4  cursor-pointer -ml-3.5"
          fill="none"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16.293 9.29297L12 13.586L7.70697 9.29297L6.29297 10.707L12 16.414L17.707 10.707L16.293 9.29297Z"
            fill="#5B6366"
          />
        </svg>
      </div>
    </div>
    <input
      className="outline-none block border-0 w-full mr-auto text-2xs md:text-xs false"
      disabled={false}
      onChange={[Function]}
      placeholder=""
      type="text"
      value=""
    />
    <svg
      className="w-4 h-4 hidden mr-4"
      fill="none"
      onClick={[Function]}
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 2C6.486 2 2 6.486 2 12C2 17.514 6.486 22 12 22C17.514 22 22 17.514 22 12C22 6.486 17.514 2 12 2ZM16.207 14.793L14.793 16.207L12 13.414L9.207 16.207L7.793 14.793L10.586 12L7.793 9.207L9.207 7.793L12 10.586L14.793 7.793L16.207 9.207L13.414 12L16.207 14.793Z"
        fill="#5B6366"
      />
    </svg>
  </div>
</div>
`;

exports[`Input Component should render with left home icon when passing leftIcon=home 1`] = `
<div
  className="w-full"
>
  <label
    className="block mb-2 text-2xs md:text-xs "
  >
    
  </label>
  <div
    className="flex items-center h-8 rounded-md px-3 py-4 border border-grey-shd5 hover:border-grey-shd3 group focus:border-dark-grey false"
  >
    <svg
      className="w-6 h-4 border-r border-grey-shd5 pr-2 br mr-4 font-light"
      fill="none"
      stroke="currentColor"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M3 12l2-2m0 0l7-7 7 7M5 10v10a1 1 0 001 1h3m10-11l2 2m-2-2v10a1 1 0 01-1 1h-3m-6 0a1 1 0 001-1v-4a1 1 0 011-1h2a1 1 0 011 1v4a1 1 0 001 1m-6 0h6"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
      />
    </svg>
    <input
      className="outline-none block border-0 w-full mr-auto text-2xs md:text-xs false"
      disabled={false}
      onChange={[Function]}
      placeholder=""
      type="text"
      value=""
    />
    <svg
      className="w-4 h-4 hidden mr-4"
      fill="none"
      onClick={[Function]}
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 2C6.486 2 2 6.486 2 12C2 17.514 6.486 22 12 22C17.514 22 22 17.514 22 12C22 6.486 17.514 2 12 2ZM16.207 14.793L14.793 16.207L12 13.414L9.207 16.207L7.793 14.793L10.586 12L7.793 9.207L9.207 7.793L12 10.586L14.793 7.793L16.207 9.207L13.414 12L16.207 14.793Z"
        fill="#5B6366"
      />
    </svg>
  </div>
</div>
`;

exports[`Input Component should render with select dropdown with its options when passing dropDownPrefix & dropDownOptions=[1,2,3] 1`] = `
<div
  className="w-full"
>
  <label
    className="block mb-2 text-2xs md:text-xs "
  >
    
  </label>
  <div
    className="flex items-center h-8 rounded-md px-3 py-4 border border-grey-shd5 hover:border-grey-shd3 group focus:border-dark-grey false"
  >
    <div
      className="border-r border-grey-shd5 pr-2 br mr-4 flex items-center"
    >
      <label
        className="text-body-small text-dark-grey capitalize hidden"
      >
        :
      </label>
      <div
        className="flex items-center"
      >
        <select
          className="relative z-10 appearance-none border-none bg-transparent pr-4 m-0 w-full font-sans text-body-small capitalize cursor-pointer outline-none"
        >
          <option
            value={1}
          >
            1
          </option>
          <option
            value={2}
          >
            2
          </option>
          <option
            value={3}
          >
            3
          </option>
          <option
            value={4}
          >
            4
          </option>
        </select>
        <svg
          className="w-4 h-4  cursor-pointer -ml-3.5"
          fill="none"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16.293 9.29297L12 13.586L7.70697 9.29297L6.29297 10.707L12 16.414L17.707 10.707L16.293 9.29297Z"
            fill="#5B6366"
          />
        </svg>
      </div>
    </div>
    <input
      className="outline-none block border-0 w-full mr-auto text-2xs md:text-xs false"
      disabled={false}
      onChange={[Function]}
      placeholder=""
      type="text"
      value=""
    />
    <svg
      className="w-4 h-4 hidden mr-4"
      fill="none"
      onClick={[Function]}
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 2C6.486 2 2 6.486 2 12C2 17.514 6.486 22 12 22C17.514 22 22 17.514 22 12C22 6.486 17.514 2 12 2ZM16.207 14.793L14.793 16.207L12 13.414L9.207 16.207L7.793 14.793L10.586 12L7.793 9.207L9.207 7.793L12 10.586L14.793 7.793L16.207 9.207L13.414 12L16.207 14.793Z"
        fill="#5B6366"
      />
    </svg>
  </div>
</div>
`;

exports[`Input Component should render with x icon to right when passing error 1`] = `
<div
  className="w-full"
>
  <label
    className="block mb-2 text-2xs md:text-xs "
  >
    
  </label>
  <div
    className="flex items-center h-8 rounded-md px-3 py-4 border border-error hover:border-grey-shd3 group focus:border-dark-grey false"
  >
    <input
      className="outline-none block border-0 w-full mr-auto text-2xs md:text-xs false"
      disabled={false}
      onChange={[Function]}
      placeholder=""
      type="text"
      value=""
    />
    <svg
      className="w-4 h-4 hidden mr-4"
      fill="none"
      onClick={[Function]}
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 2C6.486 2 2 6.486 2 12C2 17.514 6.486 22 12 22C17.514 22 22 17.514 22 12C22 6.486 17.514 2 12 2ZM16.207 14.793L14.793 16.207L12 13.414L9.207 16.207L7.793 14.793L10.586 12L7.793 9.207L9.207 7.793L12 10.586L14.793 7.793L16.207 9.207L13.414 12L16.207 14.793Z"
        fill="#5B6366"
      />
    </svg>
    <svg
      className="w-4 h-4 text-error"
      fill="none"
      stroke="currentColor"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M10 14l2-2m0 0l2-2m-2 2l-2-2m2 2l2 2m7-2a9 9 0 11-18 0 9 9 0 0118 0z"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
      />
    </svg>
  </div>
</div>
`;
